pkg/driver/controller.go

```diff
@@ -109,93 +98,85 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	}
 
 	volumeParams := req.GetParameters()
-	volumeType := volumeParams[volumeParamsVolumeType]
 
-	var storageCapacity *int64
+	err := didUserDefineDriverProvidedParameters(volumeParams)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	var tags []map[string]string
+	_, ok := volumeParams[volumeParamsTags]
+	if ok {
+		err = util.ConvertJsonStringToObject(volumeParams[volumeParamsTags], &tags)
+
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, ErrIncorrectlyFormatted, volumeParamsTags, err)
+		}
+	}
+	tags = append(tags, map[string]string{"Key": AwsFsxOpenZfsDriverTagKey, "Value": "true"})
 
+	var deleteParams = make(map[string]any)
+	for key, value := range volumeParams {
+		if strings.HasSuffix(key, "OnDeletion") {
```

Comment:
Is there any reason why we aren't as explicit as possible here? 
also nit: OnDeletion probably warrants its own constant if we go this route